server/mscalendar/welcomer.go

```diff
@@ -0,0 +1,122 @@
+package mscalendar
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/bot"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+type Welcomer interface {
+	Welcome(userID string) error
+	AfterSuccessfullyConnect(userID, userLogin string) error
+	AfterDisconnect(userID string) error
+	WelcomeFlowEnd(userID string)
+}
+
+type MSCalendarBot struct {
+	bot.Bot
+	Env
+	pluginURL string
+}
+
+const WelcomeMessage = `### Welcome to the Microsoft Calendar plugin!
+Here is some info to prove we got you logged in
+- Name: %s
+`
+
+func (m *mscalendar) Welcome(userID string) error {
+	return m.Welcomer.Welcome(userID)
+}
+
+func GetMSCalendarBot(bot bot.Bot, env Env, pluginURL string) *MSCalendarBot {
+	return &MSCalendarBot{
+		Bot:       bot,
+		Env:       env,
+		pluginURL: pluginURL,
+	}
+}
+
+func (bot *MSCalendarBot) Welcome(userID string) error {
+	bot.cleanPostIDs(userID)
+
+	postID, err := bot.DMWithAttachments(userID, bot.getConnectAttachment())
+	if err != nil {
+		return err
+	}
+
+	bot.Store.StoreUserWelcomePost(userID, postID)
+
+	return nil
+}
+
+func (bot *MSCalendarBot) AfterSuccessfullyConnect(userID, userLogin string) error {
+	postID, _ := bot.Store.DeleteUserWelcomePost(userID)
```

Comment:
Why ignore the error here?

Reply:
(1/5) I consider this error not relevant. Among other things, the error may come because there is no Welcome Post stored. My only concern is whether I get a valid postID. I could log the error just in case, but after getting to this point, I do not think we should make the function fail due to a database inconsistency.